src/libtriton/ast/bitwuzla/tritonToBitwuzlaAst.cpp

```diff
@@ -139,14 +139,19 @@ namespace triton {
 
         case BV_NODE: {
           auto childNodes = node->getChildren();
-          auto value = reinterpret_cast<IntegerNode*>(childNodes[0].get())->getInteger().convert_to<std::string>();
-          auto size = reinterpret_cast<IntegerNode*>(childNodes[1].get())->getInteger().convert_to<size_t>();
-
-          auto sort = bv_sorts.find(size);
+          auto bv_size = reinterpret_cast<IntegerNode*>(childNodes[1].get())->getInteger().convert_to<size_t>();
+          auto sort = bv_sorts.find(bv_size);
           if (sort == bv_sorts.end()) {
-            sort = bv_sorts.insert(std::make_pair(size, bitwuzla_mk_bv_sort(bzla, size))).first;
+            sort = bv_sorts.insert({bv_size, bitwuzla_mk_bv_sort(bzla, bv_size)}).first;
+          }
+
+          // Handle bitvector value as integer if it small enough.
+          if (bv_size <= sizeof(uint64_t) * 8) {
```

Comment:
I would have said `sizeof(void*)`? (what if the host is 32bit?)

Reply:
Bitwuzla [takes](https://github.com/bitwuzla/bitwuzla/blob/76865c90622e29e7ff94bb8725a0ed8f4e35332e/src/api/c/bitwuzla.h#L2615) uint64_t on both architectures.

Reply:
@JonathanSalwan I think it should work on 32bit host too because it have the same data type.